CHANGES.md

```diff
@@ -1,4 +1,183 @@
+    enabled with the new `--enable-external-profiles` flag
+  * **Breaking change** Removed the `--api-port` flag from the `inject` and
+    `install` commands, since there's no benefit to running the control plane's
+    destination API on a non-default port (thanks, @paranoidaditya)
+  * **Breaking change** Removed the `--tls=optional` flag from the
+    `linkerd install` command, since TLS is now enabled by default
+  * Changed `install` to accept or generate an issuer Secret for the Identity
+    controller
+  * Changed `install` to fail in the case of a conflict with an existing
+    installation; this can be disabled with the `--ignore-cluster` flag
+  * Added the ability to adjust the Prometheus log level
+  * Implemented `--proxy-cpu-limit` and `--proxy-memory-limit` for setting the
+    proxy resources limits (`--proxy-cpu` and `--proxy-memory` were deprecated in
+    favor of `proxy-cpu-request` and `proxy-memory-request`) (thanks @TwinProduction!)
+  * Added a validator for the `--proxy-log-level` flag
+  * Updated the `inject` and `uninject` subcommands to issue warnings when
+    resources lack a `Kind` property (thanks @Pothulapati!)
+  * The `inject` command proxy options are now converted into config
+    annotations; the annotations ensure that these configs are persisted in
+    subsequent resource updates
+  * Changed `inject` to require fetching a configuration from the control plane;
+    this can be disabled with the `--ignore-cluster` and `--disable-identity`
+    flags, though this will prevent the injected pods from participating in mesh
+    identity
+  * Included kubectl version check as part of `linkerd check` (thanks @yb172!)
+  * Updated `linkerd check` to ensure hint URLs are displayed for RPC checks
+  * Fixed sporadic (and harmless) race condition error in `linkerd check`
+  * Introduced a check for NET_ADMIN in `linkerd check`
+  * Fixed permissions check for CRDs
+  * Updated the `linkerd dashboard` command to serve the dashboard on a fixed
+    port, allowing it to leverage browser local storage for user settings
+  * Updated the `linkerd routes` command to display rows for routes that are not
+    receiving any traffic
+  * Added TCP stats to the stat command, under the `-o wide` and `-o json` flags
+  * The `stat` command now always shows the number of open TCP connections
+  * Removed TLS metrics from the `stat` command; this is in preparation for
+    surfacing identity metrics in a clearer way
+  * Exposed the `install-cni` command and its flags, and tweaked their descriptions
+  * Eliminated false-positive vulnerability warnings related to go.uuid
+* Controller
+  * Added a new public API endpoint for fetching control plane configuration
+  * **Breaking change** Removed support for running the control plane in
+    single-namespace mode, which was severely limited in the number of features
+    it supported due to not having access to cluster-wide resources
```

Comment:
Can we add a note about how we'll address this use in a better way in a later release?